### PR TITLE
Move `set_blosc!` methods to H5ZBlosc.jl

### PR DESCRIPTION
**Pull request by topolarity**
That's the last one gone!

As far as I can tell, Requires was not necessary here since the package it was "extending" already depends on HDF5.jl

### CODE REVIEW

**Comment by mkitti** [comment]
I need to think about the interaction with #1160. These methods were mainly there was backwards compatibility.

**Comment by topolarity** [comment]
For what it's worth, the methods are still there - I think this should behave pretty much as it did before, except that because we can't set retroactive `[compat]` bounds, it's possible to have, e.g., an old version of H5Zblosc with the new HDF5.jl that will be missing the methods.

**Comment by mkitti** [comment]
@simonbyrne @musm any thoughts about this pull request? Does this require a version bump?

File: Project.toml [hunk 1]
```diff
@@ -22,6 +22,7 @@ MPIExt = "MPI"
 
 [compat]
 Compat = "3.1.0, 4"
+H5Zblosc = "≥ 0.1.3"
```
Comment (simonbyrne) [inline Project.toml:25]: Shouldn't this be covered by the compat in H5Zblosc?

**Review by simonbyrne — APPROVED** [review]
I don't _think_ it should be breaking, but I guess it could break version resolution?

**Pull request merged**